docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+
+## Terminology
+
+A _widget_ is an element in a WordPress site that displays some content. Widgets generally display the same content regardless of what post or page the user is viewing.
+
+A _widget area_ is an area in a WordPress site where widgets can be placed. The active theme defines how many widget areas there are, where they appear, and what they are called.
+
+Regular widgets can only be placed **once** into a widget area. A _multi widget_, however, can be placed several times into a widget area. When a multi widget is used several times, each usage is called an _instance_.
+
+Because many WordPress themes place widgets into a sidebar column, widget areas are often referred to as _sidebars_ or _dynamic sidebars_.
+
+**For more terminology and a detailed overview of how widgets _currently_ work in WordPress, see [#14182][14182].**
+
+[14182]: https://github.com/WordPress/gutenberg/issues/14182
+
+## Prior reading
+
+- [#14182][14182] is the issue which tracks adding an API for reading and updating blocks in widget areas.
+- [#13204][13204] contains design discussion about showing blocks in `wp-admin/widgets.php`.
+- [#13205][13205] contains design discussion about showing blocks in the Customizer.
+- [#13511][13511] is a PR which added support for rendering existing widgets as a block in Gutenberg.
+
+[13204]: https://github.com/WordPress/gutenberg/issues/13204
+[13205]: https://github.com/WordPress/gutenberg/issues/13205
+[13511]: https://github.com/WordPress/gutenberg/issues/13511
+
+## Frontend
+
+A new `@wordpress/edit-widgets` package contains the UI for the `wp-admin/widgets.php` screen. This is analogous to how `@wordpress/edit-post` contains the UI for `wp-admin/post-new.php`. 
+
+Both WP Admin and the Customizer use the existing [`BlockEditor`][block-editor] component, which provides a generic UI for working with blocks, to display an interface for working with blocks in widget-editing areas.
+
+A new REST API, described in the next section, allows fetching and updating the blocks in a widget area. `@wordpress/edit-widgets` uses this API to fetch blocks which are passed to `BlockEditor`, and to update blocks when `BlockEditor` indicates that changes have been made by the user.
+
+```js
+function WidgetArea( { id } ) {
+	return (
+		<BlockEditor
+			// Illustrative only: apiGet() is not a real function. In production,
+			// @wordpress/core-data would be used to interface with the REST API.
+			blocks={ apiGet( `/wp/v2/widget-areas/${ id }` ) }
+			onChange={ ( blocks ) => {
+				// Illustrative only: apiPut() is not a real function. In production,
+				// @wordpress/core-data would be used to interface with the REST API.
+				apiPut( `/wp/v2/widget-areas/${ id }`, blocks );
+			} }
+		/>
+	);
+}
+```
+
+[block-editor]: https://github.com/WordPress/gutenberg/tree/master/packages/block-editor
+
+## REST API
+
+Fetching and updating the blocks in a widget area is done via new REST API endpoints.
+
+### The widget area resource
```

Comment:
Unsure what colour to paint this bike shed. Some previous discussion here: https://github.com/WordPress/gutenberg/pull/14612#discussion_r270247827

I see three options for the name:

- **Sidebar** — this matches what we call these areas in the WordPress Core code, e.g. `register_sidebar()`, `wp_get_sidebars_items()`, `dynamic_sidebar()`.
- **Widget Area** — this matches what we call these areas in WordPress documentation. It's slightly more accurate since widgets don't have to appear in a site's sidebar.
- **Block Area** — this matches what @mapk has been calling these areas in [his designs](https://github.com/WordPress/gutenberg/issues/13204). I do like the idea of getting rid of 'widget' as a term. (What's a widget?)

Reply:
Oh, just saw this and I commented about something similar above. To repeat my thoughts, I think we may have to keep the current terminology as we're still in a transition period and in the future phases, these areas might not be needed anymore and will move to block templates.

Reply:
Makes sense. I think it's fine if we go with 'widget areas' for now and revisit before merging into Core. The endpoint will be marked as experimental so there's no reason we can't rename in the last minute.

Reply:
Are these two different areas going to co-exist during the transition period? Would the user see both "Appearance -> Widgets" and "Appearance -> Block Areas" ? I don't think we should keep the name "Widgets" when we're showing a new interface that allows adding _any_ block, not just widgets.

I also think maybe we can be more proactive here. If the WP install is not using widgets, can we just completely transition over to Block Areas?

Reply:
In my imagination we would show either "Appearance -> Widgets" if Gutenberg is disabled _or_ "Appearance -> Block Areas" if Gutenberg is enabled. Never both.